usaspending_api/etl/management/commands/load_submission.py

```diff
@@ -495,6 +494,129 @@ def load_file_a(submission_attributes, appropriation_data, db_cursor):
         submission_attributes.submission_id)
 
 
+def get_file_b(submission_attributes, db_cursor):
+    """
+    Get broker File B data for a specific submission.
+    This function was added as a workaround for the fact that a few agencies
+    (two, as of April, 2017: DOI and ACHP) submit multiple File B records
+    for the same object class. These "dupes", come in as the same 4 digit object
+    class code but with one of the direct reimbursable flags set to NULL.
+
+    From our perspective, this is a duplicate, because we get our D/R info from
+    the 1st digit of the object class when it's four digits.
+
+    Thus, this function examines the File B data for a given submission. If
+    it has the issue of "duplicate" object classes, it will squash the
+    offending records together so that all financial totals are reporting
+    as a single object class/program activity/TAS record as expected.
+
+    If the broker validations change to prohibit this pattern in the data,
+    this intervening function will no longer be necessary, we can go back to
+    selecting * from the broker's File B data.
+
+    Args:
+        submission_attrbitues: submission object currently being loaded
+        db_cursor: db connection info
+    """
+    submission_id = submission_attributes.broker_submission_id
+
+    # does this file B have the dupe object class edge case?
+    check_dupe_oc = (
+        'SELECT count(*) '
```

Comment:
Ah, great point. So good to have your eyes on this SQL! I've been running the `count(*)` version against the broker db frequently/obsessively, making sure that no new submissions fall into this object class edge case. It runs pretty quickly against the entire body of certified submissions, so I don't expect it to be much of a chokepoint when running for an individual submission.

The schema team agrees that agencies should not submit data this way, so let's go with the "cleansed with fire soon" line of thinking.